src/js/auth.js

```diff
@@ -14,49 +14,55 @@
  * limitations under the License.
  */
 
-/* global gapi */
-
 // Utility methods for dealing with Google auth:
-// https://developers.google.com/identity/sign-in/web/reference
+// https://developers.google.com/identity/oauth2/web/guides/use-token-model
 
-export async function checkAuthStatus() {
-  // This is set on the window in `index.html`.
-  await window.authorizeUser;
+window.tokenPromise = new Promise((resolve, reject) => {
+  window.tokenLoadOkay = resolve;
+  window.tokenLoadFail = reject;
+});
+
+let tokenClient;
+let accessToken;
 
-  return getAuthInstance().isSignedIn.get();
+const clientId = window.clientId;
+const scope = window.scope;
+
+export async function initAuthClient() {
+  tokenClient = window.google.accounts.oauth2.initTokenClient({
+      client_id: clientId,
+      scope: scope,
+      callback: (tokenResponse) => {
+        accessToken = tokenResponse.access_token;
+        window.tokenLoadOkay();
+      },
+  });
+  tokenClient.requestAccessToken();
 }
 
-export function getAuthInstance() {
-  return gapi.auth2.getAuthInstance();
+export async function initAuth() {
+  document.getElementById('gis').addEventListener('load', initAuthClient);
 }
 
-// Can only be called if the user is signed in.
-export function getAccessToken() {
-  const googleUser = getAuthInstance().currentUser.get();
-  const authResponse = googleUser.getAuthResponse(true);
-  return authResponse.access_token;
+export async function signoutAccessToken() {
+  window.google.accounts.oauth2.revoke(accessToken);
+  accessToken = null;
 }
 
-export function onSignInChange(callback) {
-  getAuthInstance().isSignedIn.listen(callback);
+export async function refreshAccessToken() {
+  tokenClient.requestAccessToken();
+  return accessToken;
```

Comment:
```suggestion
  return getAccessToken();
```

eh?  slightly more consistent but it clearly not a big deal.

Reply:
Done